server/src/models/Note.ts

```diff
@@ -1,16 +1,48 @@
+// TODO: replace incorrect values for this.
+export const OCTAVE = 7;
+
 // TODO: consideration for sharps and flats.
 export class Note {
   note: string;
+  code: number;
 
-  constructor(note: string) {
-    this.validate(note);
+  private constructor(note: string, code: number) {
     this.note = note;
+    this.code = code;
+  }
+
+  static fromString(note: string) {
+    Note.validateNote(note);
+    return new Note(note, Note.getCode(note));
+  }
+
+  static fromCode(code: number) {
+    Note.validateCode(code);
+    return new Note(Note.getNote(code), code);
+  }
+
+  private static validateNote(note: string) {
+    if (note.length !== 2) throw new Error(`Invalid note: ${note}`);
```

Comment:
Hm... maybe we could log rather than throw an error? Unless we catch it somewhere downstream? This is going to cause a timeout on client if something bad happens.

Reply:
I'd prefer to do both. If the note is malformed, there's not an obvious way to recover gracefully, so I'd rather error out than fail silently or fail with a `console.log`.

But I'll definitely add a `console.log` too for easier debugging, good call.